### DISCUSSION TITLE
Add MapView Event for onRegionChangeStart

### DISCUSSION THREAD

**Discussion by lxr-mwhite** [react-native-maps/react-native-maps#5143] [Feature requests]
Looking to complement the `MapView`'s `onRegionChange` and `onRegionChangeComplete` [events](https://github.com/react-native-maps/react-native-maps/blob/master/docs/mapview.md#events) with a new event for `onRegionChangeStart`.  This would be ideal to set a state or trigger an action when the map starts moving. 

`onRegionChange` is called continuously while the map is moving.
`onRegionChangeComplete` is called once when the map stops moving. 

**Suggestion**: `onRegionChangeStart` is called when the map will start moving. 

This would be helpful to address a use case like the one described in issue #5142.

**Comment by williangaspar** [comment]
This feature will be beneficial for my project.

**Comment by salah-ghanim** [comment]
merged, I can see it being useful, thanks @lxr-mwhite for the contribution
